scenarios.md

```diff
@@ -0,0 +1,190 @@
+
+* [Helm Charts][helm-registry]
+* [Singularity][singularity]
+* Car firmware updates, deployed from OCI Artifact registries
+
+## The Need for a Generalized Signing Solution
+
+This document serves as the requirements and constraints of a generalized signing solution. It focuses on the scenarios and needs, and very specifically avoids any reference to other projects or implementations. As our working group forms a consensus on the requirements, the group will then transition to a spec.
+
+
+## Key Stake Holders & Contributors
+
+As we identify the requirements and constraints, a number of key contributors will be asked to represent their requirements and constraints.
+
+> Please add companies, projects, products that you believe should be included.
+
+* Registry Cloud Operators
+  * [Azure Container Registry (acr)][acr] - Steve Lasker <steve.lasker@microsoft.com> (@stevelasker)
+  * [AWS Container Registry (ecr)][ecr] - Omar Paul <omarpaul@amazon.com>
+  * [Docker Hub][docker-hub]
+  * [Google Container Registry (gcr)][gcr]
+  * [GitHub Package Registry (gpr)][gpr]
+  * [Quay][quay]
+  * [IBM Cloud Container Registry (icr)][icr]
+* Registry Vendors, Projects & Products
+  * [Docker Trusted Registry][docker-dtr]
+  * [Harbor][harbor]
+  * [JFrog Artifactory][jfrog]
+* Artifact Types
+  * Container Images - OCI Org
+  * Helm Charts
+  * Singularity
+  * Operator Bundles
+
+## Scenarios
+
+### Scenario #1: Local Build, Sign, Validate
+
+Prior to doing any deployment, a developer can test the: build, sign, validate scenario.
+
+1. Locally build a container image using a non-registry specific `name:tag`, such as:  
+  `$docker build hello-world:1`
+1. Locally sign `hello-world:1` 
+2. Run the image in a local environment which was configured to only support signed artifacts. *(**note** - details on the usability is deferred from this requirement. This could be a local docker host, a local kind or minikube instance configured to only run signed images)*  
+  `$ docker run hello-world:1`
+
+**Implications of this requirement:**
+
+- The local developer has access to the signing keys
+- The local environment has a policy by which it states the set of signatures it accepts.
+- The signing and validation of artifacts does not require a registry.
+- The signature used for validation may be hosted in a registry, or other accessible location.
+
+### Scenario #2: Sign, Rename, Push, Validate
+
+Once the developer has locally validated the build, sign, validate scenario, they will likely want to push the artifact to a registry used for deployment.
+
+1. Locally build and sign an artifact, such as a container image
+1. Rename the artifact to reflect the registry it will be pushed to:  
+  `$ docker tag hello-world:1 myregistry.domain.com/marketing/web:1`
+  `$docker push myregistry.domain.com/marketing/web:1`
+1. Deploy the artifact to a cluster that requires signatures:  
+  `$kubectl apply -f kubedeploy.yaml`
```

Comment:
I agree with the generalized statement here. What I'm trying to suggest is we do want/need integration from container, or any artifact host that retrieves artifacts and supports signing verification.
If we generalize this to pull, are we overly focusing on the docker client, and missing the scenarios where artifact hosts interact with registries directly? Or, using CRI. 
Suggestions?

Reply:
> are we overly focusing on the docker client, and missing the scenarios where artifact hosts interact with registries directly?

I think we want to design the spec in such a way that any interested client is able to integrate, including container runtimes that pull images (like Docker), orchestrators attempting to apply something like an admission or trust policy, scanning systems, etc.

The examples right now are pretty focused on Docker.  That's somewhat reasonable, as we expect most of the interested parties to be at least familiar with how to use Docker today.  But I don't think adding Kubernetes or CRI here makes this particular scenario better (and as someone who works on a non-Kubernetes orchestrator that doesn't use CRI), I'd prefer to not see those enshrined as the default way to do something.

It might be worth adding a separate scenario covering things other than container runtimes, and using a scanning service or an orchestrator as an example.

Reply:
I removed references to specific orchestrators or other projects.